core-web/src/main/java/org/silverpeas/core/web/util/viewgenerator/html/pagination/PaginationSP.java

```diff
@@ -114,29 +128,56 @@ public String printIndex(String javascriptFunc) {
         int index = getIndexForNextPage();
-        result.append("<div class=\"pageOff\">");
+        result.append("<a class=\"pageOff\"");
         result.append(getLink(javascriptFunc, index, getString("GEF.pagination.nextPage")));
         result.append("<img src=\"").append(getIconsPath()).append(
-            "/arrows/arrowRight.gif\" border=\"0\" align=\"absmiddle\" alt=\"")
-            .append(getString("GEF.pagination.nextPage")).append("\"/></a>");
-        result.append("</div>");
+            "/arrows/arrowRight.png\" border=\"0\" align=\"absmiddle\" alt=\"")
+            .append(getString("GEF.pagination.nextPage")).append("\"/>");
+        result.append("</a>");
 
         if (currentPage + getNumberOfPagesAround() < nbPage) {
           // display last page link
-          result.append("<div class=\"pageOff\">");
+          result.append("<a class=\"pageOff\"");
           result.append(getLink(javascriptFunc, getIndexForLastPage(),
               getString("GEF.pagination.lastPage")));
           result.append("<img src=\"").append(getIconsPath()).append(
-              "/arrows/arrowDoubleRight.gif\" border=\"0\" align=\"absmiddle\" alt=\"")
-              .append(getString("GEF.pagination.lastPage")).append("\"/></a> ");
-          result.append("</div>");
+              "/arrows/arrowDoubleRight.png\" border=\"0\" align=\"absmiddle\" alt=\"")
+              .append(getString("GEF.pagination.lastPage")).append("\"/>");
+          result.append("</a>");
         }
       }
+      result.append("</div>");
 
+      if (nbItemsPerPage && getNbItems() > 25) {
+        result.append("<div class=\"pageIndex numberPerPage\">");
+        int[] values = new int[]{25, 50, 100};
+        for (int i=0; i<values.length; i++) {
+          int value = values[i];
+          if (getNbItems() > value || (i != 0 && getNbItems() > values[i-1])) {
+            if (getNbItemsPerPage() == value) {
+              result.append("<a class=\"selected\">").append(value).append("</a>");
+            } else {
+              result.append("<a href=\"").append(getNbItemsPerPageLink(javascriptFunc, value))
+                  .append("\" title=\"")
+                  .append(getStringWithParam("GEF.pagination.perPage", String.valueOf(value))).append("\">")
+                  .append(value).append("</a>");
+            }
+          }
+        }
+        // add special feature : All
+        int specialValue = 100000;
+        if (getNbItemsPerPage() == specialValue) {
+          result.append("<a class=\"selected\">").append(getString("GEF.pagination.all")).append("</a>");
+        } else {
+          result.append(
+              "<a href=\"").append(getNbItemsPerPageLink(javascriptFunc, specialValue)).append("\" title=\"").append(getString("GEF.pagination.all.title")).append("\">").append(getString("GEF.pagination.all")).append("</a>");
+        }
+        result.append("</div>");
+      }
+
+      boolean displayJumper = nbPage > 12;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Assign this magic number 12 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS109)